src/clib/pio.h

```diff
@@ -48,9 +48,17 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
+#include <limits.h>
 #include <adios2_c.h>
-#define _ADIOS_ALL_PROCS 1 /* ADIOS: assume all procs are also IO tasks */
-#define ADIOS_PIO_MAX_DECOMPS 200 /* Maximum number of decomps */
+
+#define ADIOS_PIO_MAX_DECOMPS 1024 /* Maximum number of decomps */
```

Comment:
An upper limit on the number of decompositions can be an issue in the future (#449).
We don't need to fix this issue in this PR, but something to be taken care of in the future.